src/lib/onHidden.ts

```diff
@@ -14,18 +14,10 @@
  * limitations under the License.
  */
 
-export interface OnHiddenCallback {
-  (event: Event): void;
-}
-
-export const onHidden = (cb: OnHiddenCallback) => {
-  const onHiddenOrPageHide = (event: Event) => {
-    if (event.type === 'pagehide' || document.visibilityState === 'hidden') {
-      cb(event);
+export const onHidden = (cb: () => void) => {
+  document.addEventListener('visibilitychange', () => {
+    if (document.visibilityState === 'hidden') {
+      cb();
     }
-  };
-  addEventListener('visibilitychange', onHiddenOrPageHide, true);
-  // Some browsers have buggy implementations of visibilitychange,
-  // so we use pagehide in addition, just to be safe.
-  addEventListener('pagehide', onHiddenOrPageHide, true);
```

Comment:
Do we feel this is no longer the case? Is that why you removed this?

Reply:
Yeah, TBH I've wanted to remove this for a long time.

Last I checked (when working on the PendingBeacon polyfill 1.5 years ago) all bugs I was aware of have been fixed. And for old browsers that still have these bugs, that primarily affects Safari (which doesn't support the metrics that this is needed for) and old Chrome (which also doesn't support these metrics), so I think it's 100% safe to remove this.

It's also potentially racy in the sense that `document.visibilityState` will show as "visible" in the pagehide event, so the logic added to this PR that checks visibility state first and then calls `onHidden()` could break in a browser that doesn't reliably support `visibilitychange` anyway.

There's also the issue that if your app's code is doing something on `visibilitychange` and then you're making some call to the web-vitals library, it would be too late (e.g. web-vitals code already ran on `pagehide`).